backend/src/main/java/com/github/astridottmann/backend/controllers/MongoUserController.java

```diff
@@ -32,4 +39,17 @@ public void logout(HttpSession httpSession) {
         httpSession.invalidate();
         SecurityContextHolder.clearContext();
     }
+
+    @PostMapping("/signin")
+    public MongoUser signIn(@RequestBody @Valid MongoUser user) {
+        String errorMessage = "Username already exists!";
+
+        if (mongoUserRepository.findMongoUserByUsername(user.username()).isPresent()) {
+            throw new IllegalArgumentException(errorMessage);
+        }
```

Comment:
You could put the errorMessage inside the if. You only create an instance when you need it.

```suggestion
        if (mongoUserRepository.findMongoUserByUsername(user.username()).isPresent()) {
            String errorMessage = "Username already exists!";
            throw new IllegalArgumentException(errorMessage);
        }
```